CPT331.iOS/CPT331.iOS/MapViewController.swift

```diff
@@ -0,0 +1,376 @@
+        searchResultsTable.tableFooterView = UIView()
+        searchResultsTable.delegate = self
+        searchResultsTable.dataSource = self
+        
+        // Setup map
+        mapView.delegate = self
+        
+        // Register double tap recognizer so the single tap knows to ignore them
+        let doubleTap = UITapGestureRecognizer(target: self, action: nil)
+        doubleTap.numberOfTapsRequired = 2
+        self.mapView.addGestureRecognizer(doubleTap)
+
+        // Register single tap recognizer to check if the user tapped on a city/town/village label
+        let singleTap = UITapGestureRecognizer(target: self, action: #selector(self.mapSingleTapped))
+        singleTap.requireGestureRecognizerToFail(doubleTap)
+        singleTap.delegate = self
+        singleTap.cancelsTouchesInView = false
+        self.mapView.addGestureRecognizer(singleTap)
+        
+        // Bind observers to listen to keyboard show/hide events
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(MapViewController.keyboardWillShow(_:)), name: UIKeyboardWillShowNotification, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(MapViewController.keyboardWillHide(_:)), name: UIKeyboardWillHideNotification, object: nil)
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        if segue.identifier == "showLocationView", let location = self.lastLocationTapped {
+            let vc = segue.destinationViewController as! LocationViewController
+            vc.location = location
+            
+        } else if segue.identifier == "showEventView", let event = self.lastEventTapped {
+            let vc = segue.destinationViewController as! EventViewController
+            vc.event = event
+        }
+    }
+    
+    
+    
+    
+    /* --------------------------------- *
+     *         Map Functionality         *
+     * --------------------------------- */
+    func getRadius(fromCoordinateBounds bounds: MGLCoordinateBounds) -> Double {
+        let ne = bounds.ne
+        let sw = bounds.sw
+        
+        var point1: CLLocationCoordinate2D!
+        var point2: CLLocationCoordinate2D!
+        
+        // Use longitude if in portrait
+        if (sw.longitude - ne.longitude) >= (sw.latitude - ne.latitude) {
+            point1 = CLLocationCoordinate2D(latitude: 0, longitude: sw.longitude)
+            point2 = CLLocationCoordinate2D(latitude: 0, longitude: ne.longitude)
+            
+        // Otherwise, use latitude if in landscape
+        } else {
+            point1 = CLLocationCoordinate2D(latitude: sw.latitude, longitude: 0)
+            point2 = CLLocationCoordinate2D(latitude: ne.latitude, longitude: 0)
+        }
+        
+        // Earth's radius in Kilometers
```

Comment:
Agreed. Similar code in 3 files:
* CPT331.iOS/CPT331.iOS/CDSuburb.swift
* CPT331.iOS/CPT331.iOS/MapViewController.swift
*  CPT331.iOS/CPT331.iOS/GeocodedPlacemark.swift

This might justify a class on its own.